In [2]:
context = (
    "The intergovernmental Panel on Climate Change (IPCC) is a scientific intergovernmental body under the auspices "
    "of the United Nations, set up at the request of member governments. It was first established in 1988 by two United "
    "Nations organizations, the World Meteorological Organization (WMO) and the United Nations Environment Programme "
    "(UNEP), and later endorsed by the United Nations General Assembly through Resolution 43/53. Membership of the IPCC "
    "is open to all members of the WMO and UNEP. The IPCC produces reports that support the United Nations framework "
    "Convention on Climate Change (UNFCCC), which is the main international treaty on climate change. The ultimate "
    "objective of the UNFCCC is to \"stabilize greenhouse gas concentrations in the atmosphere at a level that would "
    "prevent dangerous anthropogenic [i.e., human-induced] interference with the climate system\". IPCC reports cover "
    "\"the scientific, technical and socio-economic information relevant to understanding the scientific basis of risk "
    "of human-induced climate change, its potential impacts and options for adaptation and mitigation.\""
)

context

'The intergovernmental Panel on Climate Change (IPCC) is a scientific intergovernmental body under the auspices of the United Nations, set up at the request of member governments. It was first established in 1988 by two United Nations organizations, the World Meteorological Organization (WMO) and the United Nations Environment Programme (UNEP), and later endorsed by the United Nations General Assembly through Resolution 43/53. Membership of the IPCC is open to all members of the WMO and UNEP. The IPCC produces reports that support the United Nations framework Convention on Climate Change (UNFCCC), which is the main international treaty on climate change. The ultimate objective of the UNFCCC is to "stabilize greenhouse gas concentrations in the atmosphere at a level that would prevent dangerous anthropogenic [i.e., human-induced] interference with the climate system". IPCC reports cover "the scientific, technical and socio-economic information relevant to understanding the scientific ba

In [8]:
questions = [
    "What organization is the IPCC a part of?",
    "What UN organizations established the IPCC?",
    "What does the UN want to stabilize?"
]

https://huggingface.co/models

deepset/bert-base-cased-squad2
deepset/electra-base-squad2

In [3]:
from transformers import BertForQuestionAnswering

In [4]:
model = BertForQuestionAnswering.from_pretrained('deepset/bert-base-cased-squad2')

In [5]:
from transformers import AutoTokenizer

In [6]:
tokenizer = AutoTokenizer.from_pretrained('deepset/bert-base-cased-squad2')

In [9]:
tokenizer.encode(questions[0], truncation=True, padding=True)

[101, 1327, 2369, 1110, 1103, 14274, 12096, 170, 1226, 1104, 136, 102]

101 = [CLS] = start of sequence
102 = [SEP] = separator

"[CLS] <context> [SEP] <question> [SEP] [PAD] [PAD]"

In [10]:
tokenizer.encode('[CLS]')

[101, 101, 102]

In [11]:
from transformers import pipeline

In [12]:
nlp = pipeline('question-answering', model=model, tokenizer=tokenizer)

In [19]:
nlp({
    'question': questions[0],
    'context': context
})

{'score': 0.450367271900177,
 'start': 118,
 'end': 132,
 'answer': 'United Nations'}

In [16]:
context[118:132]

'United Nations'